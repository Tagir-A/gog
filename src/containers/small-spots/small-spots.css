.small-spots {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}
