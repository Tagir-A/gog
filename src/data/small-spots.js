import oddworld from '../assets/small-spots/oddworld.png'
import deponia from '../assets/small-spots/deponia.png'
import theSettlers2 from '../assets/small-spots/the-settlers-2.png'
import neverwinterNights from '../assets/small-spots/neverwinter-nights.png'
import assasinsCreed from '../assets/small-spots/assasins-creed.png'

const SMALL_SPOTS = [{
        title: "ODDWORLD: STRANGER'S WRATH",
        img: oddworld,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        },
        discount: 50
    },
    {
        title: "CHAOS ON DEPONIA",
        img: deponia,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        },
        owned: true
    },
    {
        title: "THE SETTLERS 2: GOLD EDITION",
        img: theSettlers2,
        price: {
            currency: 'USD',
            value: 599,
            minorUnits: 100
        }
    },
    {
        title: "NEVERWINTER NIGHTS",
        img: neverwinterNights,
        price: {
            currency: 'USD',
            value: 499,
            minorUnits: 100
        },
        discount: 50
    },
    {
        title: "ASSASIN'S CREED: DIRECTOR'S CUT",
        img: assasinsCreed,
        price: {
            currency: 'USD',
            value: 999,
            minorUnits: 100
        }
    }
]

export {
    SMALL_SPOTS
}
