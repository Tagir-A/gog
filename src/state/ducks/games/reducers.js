import {
    SMALL_SPOTS
} from "../../../data/small-spots";

const defaultState = [...SMALL_SPOTS]


export default function games(state = defaultState, action = {}) {
    return state;
}
