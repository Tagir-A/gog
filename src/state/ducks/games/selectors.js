export function isInCart(game, {
    products
}) {
    return Boolean(products.find(product => product.title === game.title));
}
