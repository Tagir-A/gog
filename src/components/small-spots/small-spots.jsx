import React, { Component } from "react";
import { PropTypes } from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import "./small-spots.css";

import { smallSpots } from "../../data/small-spots";
import { SmallSpot } from "../small-spot/small-spot";
import { addToCart } from "../../actions/cart";

function isInCart(game, { products }) {
  return products.find(product => product.title === game.title);
}

class SmallSpots extends Component {
  static propTypes = {
    cart: PropTypes.shape({
      products: PropTypes.shape({
        title: PropTypes.string.isRequired
      })
    }),
    addToCart: PropTypes.func.isRequired
  };

  render() {
    return (
      <section className="small-spots">
        {smallSpots.map(game => (
          <SmallSpot
            key={game.title}
            game={game}
            isInCart={isInCart(game, this.props.cart)}
            onAddClick={this.props.addToCart}
          />
        ))}
      </section>
    );
  }
}

function mapState({ cart }) {
  return {
    cart
  };
}

function mapDispatch(dispatch) {
  return bindActionCreators(
    {
      addToCart
    },
    dispatch
  );
}

export default connect(mapState, mapDispatch)(SmallSpots);
