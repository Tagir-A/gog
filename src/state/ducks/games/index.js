import reducer from "./reducers";
import * as gamesSelectors from "./selectors";

export {
    gamesSelectors
}
export default reducer;
