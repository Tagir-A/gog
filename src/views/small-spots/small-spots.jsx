import React, { Component } from "react";
import { PropTypes } from "prop-types";
import { SmallSpot } from "../small-spot/small-spot";
import "./small-spots.css";
import game from "../../models/game";

function SmallSpots({ games, addToCart }) {
  return (
    <section className="small-spots">
      {games.map(game => (
        <SmallSpot key={game.title} game={game} onAddClick={addToCart} />
      ))}
    </section>
  );
}

SmallSpots.propTypes = {
  cart: PropTypes.shape({
    products: PropTypes.arrayOf(game)
  }),
  games: PropTypes.arrayOf(game),
  addToCart: PropTypes.func.isRequired
};

export { SmallSpots };
