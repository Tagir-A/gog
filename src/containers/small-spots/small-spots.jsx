import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { SmallSpots } from "../../views/small-spots/small-spots";

import { cartActions } from "../../state/ducks/cart";
import { gamesSelectors } from "../../state/ducks/games";

function mapState({ cart, games }) {
  const { isInCart } = gamesSelectors;
  return {
    cart,
    games: games.map(game => ({ ...game, isInCart: isInCart(game, cart) }))
  };
}

function mapDispatch(dispatch) {
  const { addToCart } = cartActions;
  return bindActionCreators(
    {
      addToCart
    },
    dispatch
  );
}

export default connect(mapState, mapDispatch)(SmallSpots);
