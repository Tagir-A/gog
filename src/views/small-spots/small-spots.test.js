import { SmallSpots } from "./small-spots";
import {GAMES} from '../../data/games'

describe('small-spots', () => {
    it('renders as expected', () => {
        const wrapper = shallow(<SmallSpots
            games={GAMES}
            addToCart={()=>null}
            />)
        expect(wrapper).toMatchSnapshot()
    })
})