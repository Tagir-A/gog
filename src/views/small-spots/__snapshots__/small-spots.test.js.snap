// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`small-spots renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SmallSpots
    addToCart={[Function]}
    games={
        Array [
            Object {
              "discount": 50,
              "img": "oddworld.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "ODDWORLD: STRANGER'S WRATH",
            },
            Object {
              "img": "deponia.png",
              "owned": true,
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "CHAOS ON DEPONIA",
            },
            Object {
              "img": "the-settlers-2.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 599,
              },
              "title": "THE SETTLERS 2: GOLD EDITION",
            },
            Object {
              "discount": 50,
              "img": "neverwinter-nights.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 499,
              },
              "title": "NEVERWINTER NIGHTS",
            },
            Object {
              "img": "assasins-creed.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "ASSASIN'S CREED: DIRECTOR'S CUT",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SmallSpot
          game={
                    Object {
                              "discount": 50,
                              "img": "oddworld.png",
                              "price": Object {
                                "currency": "USD",
                                "minorUnits": 100,
                                "value": 999,
                              },
                              "title": "ODDWORLD: STRANGER'S WRATH",
                            }
          }
          onAddClick={[Function]}
/>,
        <SmallSpot
          game={
                    Object {
                              "img": "deponia.png",
                              "owned": true,
                              "price": Object {
                                "currency": "USD",
                                "minorUnits": 100,
                                "value": 999,
                              },
                              "title": "CHAOS ON DEPONIA",
                            }
          }
          onAddClick={[Function]}
/>,
        <SmallSpot
          game={
                    Object {
                              "img": "the-settlers-2.png",
                              "price": Object {
                                "currency": "USD",
                                "minorUnits": 100,
                                "value": 599,
                              },
                              "title": "THE SETTLERS 2: GOLD EDITION",
                            }
          }
          onAddClick={[Function]}
/>,
        <SmallSpot
          game={
                    Object {
                              "discount": 50,
                              "img": "neverwinter-nights.png",
                              "price": Object {
                                "currency": "USD",
                                "minorUnits": 100,
                                "value": 499,
                              },
                              "title": "NEVERWINTER NIGHTS",
                            }
          }
          onAddClick={[Function]}
/>,
        <SmallSpot
          game={
                    Object {
                              "img": "assasins-creed.png",
                              "price": Object {
                                "currency": "USD",
                                "minorUnits": 100,
                                "value": 999,
                              },
                              "title": "ASSASIN'S CREED: DIRECTOR'S CUT",
                            }
          }
          onAddClick={[Function]}
/>,
      ],
      "className": "small-spots",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "ODDWORLD: STRANGER'S WRATH",
        "nodeType": "function",
        "props": Object {
          "game": Object {
            "discount": 50,
            "img": "oddworld.png",
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 999,
            },
            "title": "ODDWORLD: STRANGER'S WRATH",
          },
          "onAddClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "CHAOS ON DEPONIA",
        "nodeType": "function",
        "props": Object {
          "game": Object {
            "img": "deponia.png",
            "owned": true,
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 999,
            },
            "title": "CHAOS ON DEPONIA",
          },
          "onAddClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "THE SETTLERS 2: GOLD EDITION",
        "nodeType": "function",
        "props": Object {
          "game": Object {
            "img": "the-settlers-2.png",
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 599,
            },
            "title": "THE SETTLERS 2: GOLD EDITION",
          },
          "onAddClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "NEVERWINTER NIGHTS",
        "nodeType": "function",
        "props": Object {
          "game": Object {
            "discount": 50,
            "img": "neverwinter-nights.png",
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 499,
            },
            "title": "NEVERWINTER NIGHTS",
          },
          "onAddClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ASSASIN'S CREED: DIRECTOR'S CUT",
        "nodeType": "function",
        "props": Object {
          "game": Object {
            "img": "assasins-creed.png",
            "price": Object {
              "currency": "USD",
              "minorUnits": 100,
              "value": 999,
            },
            "title": "ASSASIN'S CREED: DIRECTOR'S CUT",
          },
          "onAddClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SmallSpot
            game={
                        Object {
                                    "discount": 50,
                                    "img": "oddworld.png",
                                    "price": Object {
                                      "currency": "USD",
                                      "minorUnits": 100,
                                      "value": 999,
                                    },
                                    "title": "ODDWORLD: STRANGER'S WRATH",
                                  }
            }
            onAddClick={[Function]}
/>,
          <SmallSpot
            game={
                        Object {
                                    "img": "deponia.png",
                                    "owned": true,
                                    "price": Object {
                                      "currency": "USD",
                                      "minorUnits": 100,
                                      "value": 999,
                                    },
                                    "title": "CHAOS ON DEPONIA",
                                  }
            }
            onAddClick={[Function]}
/>,
          <SmallSpot
            game={
                        Object {
                                    "img": "the-settlers-2.png",
                                    "price": Object {
                                      "currency": "USD",
                                      "minorUnits": 100,
                                      "value": 599,
                                    },
                                    "title": "THE SETTLERS 2: GOLD EDITION",
                                  }
            }
            onAddClick={[Function]}
/>,
          <SmallSpot
            game={
                        Object {
                                    "discount": 50,
                                    "img": "neverwinter-nights.png",
                                    "price": Object {
                                      "currency": "USD",
                                      "minorUnits": 100,
                                      "value": 499,
                                    },
                                    "title": "NEVERWINTER NIGHTS",
                                  }
            }
            onAddClick={[Function]}
/>,
          <SmallSpot
            game={
                        Object {
                                    "img": "assasins-creed.png",
                                    "price": Object {
                                      "currency": "USD",
                                      "minorUnits": 100,
                                      "value": 999,
                                    },
                                    "title": "ASSASIN'S CREED: DIRECTOR'S CUT",
                                  }
            }
            onAddClick={[Function]}
/>,
        ],
        "className": "small-spots",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "ODDWORLD: STRANGER'S WRATH",
          "nodeType": "function",
          "props": Object {
            "game": Object {
              "discount": 50,
              "img": "oddworld.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "ODDWORLD: STRANGER'S WRATH",
            },
            "onAddClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "CHAOS ON DEPONIA",
          "nodeType": "function",
          "props": Object {
            "game": Object {
              "img": "deponia.png",
              "owned": true,
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "CHAOS ON DEPONIA",
            },
            "onAddClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "THE SETTLERS 2: GOLD EDITION",
          "nodeType": "function",
          "props": Object {
            "game": Object {
              "img": "the-settlers-2.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 599,
              },
              "title": "THE SETTLERS 2: GOLD EDITION",
            },
            "onAddClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "NEVERWINTER NIGHTS",
          "nodeType": "function",
          "props": Object {
            "game": Object {
              "discount": 50,
              "img": "neverwinter-nights.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 499,
              },
              "title": "NEVERWINTER NIGHTS",
            },
            "onAddClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ASSASIN'S CREED: DIRECTOR'S CUT",
          "nodeType": "function",
          "props": Object {
            "game": Object {
              "img": "assasins-creed.png",
              "price": Object {
                "currency": "USD",
                "minorUnits": 100,
                "value": 999,
              },
              "title": "ASSASIN'S CREED: DIRECTOR'S CUT",
            },
            "onAddClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
